coalib/bearlib/languages/Language.py

```diff
@@ -410,40 +443,171 @@ def get_default_version(self):
         return type(self)(self.versions[-1]) if self.versions else type(self)()
 
 
-def limit_versions(language, limit, operator):
+def prepare_instance(language_, item):
     """
-    Limits given languages with the given operator:
+    It creates a new `Language` instance depending on the type of item. It is
+    used in __and__ , __or__ methods to avoid duplicate code.
+
+    :param item:
+        Item to use to create the new instance.
+    :return:
+        A `Language` instance.
+    """
+    if isinstance(item, float):
+        language = type(language_)((item))
+    elif isinstance(item, int):
+        language = Language[type(language_).__name__ + ' ' + str(item)]
+    elif isinstance(item, str):
+        language = Language[item]
+    else:
+        language = item
+    return language
+
+
+def limit_versions(language, specifier):
+    """
+    Limits the versions of a `Language` instance. It supports limiting by using
+    operators or specifying versions:
+
+    >>> Language['Python ~= 3']
+    Python 3.3, 3.4, 3.5, 3.6
+    >>> Language['Python 3.5, 3.6, 3.7']
+    Traceback (most recent call last):
+    ...
+    ValueError: 3.7 version does not exist
+    >>> Language['py 2']
+    Python 2.7
+    >>> Language['py < 2']
+    Traceback (most recent call last):
+    ...
+    ValueError: No versions left
 
     :param language:
         A `Language` instance.
-    :param limit:
-        A number to limit the versions.
-    :param operator:
-        The operator to use for the limiting.
+    :param specifier:
+        The specifier to use for the limiting.
     :return:
         A new `Language` instance with limited versions.
     :raises ValueError:
         If no version is left anymore.
     """
-    if isinstance(limit, int):
+    if not specifier[0].isdigit():
+        specs = SpecifierSet(specifier + '.0')
         versions = [version for version in language.versions
-                    if operator(int(version), limit)]
+                    if str(version) in specs]
     else:
-
-        versions = [version for version in language.versions
-                    if operator(version, limit)]
+        specifiers = specifier.split(', ')
+        versions = []
+        for specifier in specifiers:
+            specs = SpecifierSet('~=' + specifier + '.0')
```

Comment:
@MariosPanag And one more thing :) When handling every `specifier` after `.split(',')` then you can also simply use `packaging.specifiers.Specifier` instead of `SpecifierSet`